### PR TITLE
Fixed example with incorrect allowCredential. Improved existing examples

### PR DESCRIPTION
**Pull request by herrjemand**
***
<a href="https://pr-preview.s3.amazonaws.com/herrjemand/webauthn/pull/888.html" title="Last updated on May 24, 2018, 12:46 PM GMT (a86910e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/888/3c5e383...herrjemand:a86910e.html" title="Last updated on May 24, 2018, 12:46 PM GMT (a86910e)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -4569,13 +4569,14 @@ credentials, then the sample code for performing such an authentication might lo
 
 <pre class="example" highlight="js">
     if (!window.PublicKeyCredential) { /* Platform not capable. Handle error. */ }
-
+  
+    var idBuffer = encoder.encode("agent007_4124732214")
```
Comment (emlun) [inline index.bs]: I think I'd prefer this to be something like

```
var idBuffer = new Uint8Array([183, 148, 245 /* more random bytes previously generated by the authenticator */]);
```

analogous with the `challenge` example, so as to not indicate that the credential ID is a readable identifier chosen by the RP or user.

Comment (equalsJeffH) [reply index.bs]: I concur with @emlun's comment above, and would even add an additional comment along the lines of "credential ID ought to be an opaque and random"

**Comment by equalsJeffH** [comment]
merge from upstream/master ought to fix the breaking build

**Comment by herrjemand** [comment]
Sorry @emlun I totally got lost with work.

**Comment by herrjemand** [comment]
PR looks good to me

File: index.bs [hunk 2]
```diff
@@ -4591,12 +4591,14 @@ credentials, then the sample code for performing such an authentication might lo
...
 
+    // credentialId is generated by the authenticator and should be an opaque random byte array
+    var credentialId = new Uint8Array([183, 148, 245 /* more random bytes previously generated by the authenticator */]);
     var options = {
-                    // The challenge must be produced by the server, see the Security Considerations
-                    challenge: new Uint8Array([4,101,15 /* 29 more random bytes generated by the server */]),
-                    timeout: 60000,  // 1 minute
-                    allowCredentials: [{ type: "public-key" }]
-                  };
+      // The challenge must be produced by the server, see the Security Considerations
```
Comment (selfissued) [inline index.bs]: Change the "," to ";" so that the comment doesn't contain a run-on sentence

File: index.bs [hunk 3]
```diff
@@ -4616,21 +4618,21 @@ extension for transaction authorization.
...
 
     var options = {
-                    // The challenge must be produced by the server, see the Security Considerations
-                    challenge: new Uint8Array([8,18,33 /* 29 more random bytes generated by the server */]),
-                    timeout: 60000,  // 1 minute
-                    allowCredentials: [acceptableCredential1, acceptableCredential2],
-                    extensions: { 'txAuthSimple':
-                       "Wave your hands in the air like you just don't care" }
-                  };
+      // The challenge must be produced by the server, see the Security Considerations
```
Comment (selfissued) [inline index.bs]: Change the "," to ";" so that the comment doesn't contain a run-on sentence

**Comment by emlun** [comment]
@nadalin @equalsJeffH @selfissued Looks like there are still IPR issues that need to be resolved.

**Comment by nadalin** [comment]
@emlun please fix and merge

**Comment by emlun** [comment]
@nadalin How do I fix it? Didn't we say on the 2018-05-30 call that we would work on getting @herrjemand in as a contributor?

**Comment by wseltzer** [comment]
IPR check done. IE confirmed and revalidated.

**Pull request merged**